### PR TITLE
Added importRules option to inject @import rules for generated files

### PR DESCRIPTION
**Pull request by entozoon**
So, say your primary CSS file is app.css, it would get prepended with lines such as:
```
@import url(/css/app-blessed1.css);
@import url(/css/app-blessed2.css);
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "bless-css-webpack-plugin",
-  "version": "1.0.1",
+  "version": "1.0.2",
```
Comment (mattlewis92) [inline package.json]: Please undo this as it will be handled by the release script automatically

File: src/plugin.js [hunk 2]
```diff
@@ -35,13 +38,33 @@ class BlessCSSWebpackPlugin {
...
+               * Inject @import rules into the primary .css file for all others
+               */
+              if (this.options.importRules) {
+                const parsedDataSimple = bless.chunk(input.source);
+                let importRules = '';
+
+                parsedDataSimple.data.map((fileContents, index) => {
+                  if (index == 0) return;
+
+                  let filename = index === 0 ? cssFileName : `${filenameWithoutExtension}-blessed${index}.css`;
```
Comment (mattlewis92) [inline src/plugin.js]: This should be extracted into a shared function with the code below

**Comment by entozoon** [comment]
Hello there, good question!
As sites scale up, it's not certain how many files it will split into, if any (especially if turning features on and off), so generated imports is useful.

I'll look into making changes as requested :)

File: src/plugin.js [hunk 3]
```diff
@@ -35,13 +38,33 @@ class BlessCSSWebpackPlugin {
...
+                  //e.g. @import url(app-blessed1.css);
+                  importRules += '@import url(' + filename + ');\n';
```
Comment (mattlewis92) [inline src/plugin.js]: The filename also needs to respect the [`output.path`](https://webpack.js.org/configuration/output/#output-path) and [`output.publicPath`](https://webpack.js.org/configuration/output/#output-publicpath) webpack options.

**Comment by mattlewis92** [comment]
All the CSS files will still be requested by the browser though either way. Using `@import` will probably give you a slight drop in performance as the entry CSS file will need to be loaded first before any of the other chunks can be loaded.

Also can you [disable this](https://github.com/mattlewis92/bless-css-webpack-plugin/pull/3/files#diff-a4f485aa834ca5b3c24819fceb39c56fL57) if `importRules` is set as otherwise the imported files will still be injected into the head of the document if using the html-webpack-plugin

**Comment by entozoon** [comment]
I belieeeeve (and hope (and pray)) that I've made all the changes you desired :D

**Comment by mattlewis92** [comment]
Ah sorry found another bug with this implementation - import's need to be at the [top of the CSS file](https://developer.mozilla.org/en/docs/Web/CSS/@import) for it to work:

> The @import CSS at-rule is used to import style rules from other style sheets. These rules must precede all other types of rules

So you'll have to reverse order the CSS files and have the last file import the preceding ones before it. This will then break sourcemaps though as the line numbers won't match up. There is an [implementation](https://github.com/BlessCSS/gulp-bless/blob/master/index.js#L106-L119) in the gulp plugin for bless that can guide you. They also handle sourcemaps there as well, the vinyl-sourcemaps-apply module is too high an abstraction though as its only really for gulp, it would be better to use the `source-map` module directly.

Also the the imported filename needs to respect the [`output.path`](https://webpack.js.org/configuration/output/#output-path) and [`output.publicPath`](https://webpack.js.org/configuration/output/#output-publicpath) webpack options. `output.publicPath` is easy as its just `publicPath + '/' + filename`, `output.path` is trickier though as it can be a full filepath so you'll need to remove `process.cwd()` from it. There may be a webpack API / a module to make this easier but I can't find anything from google.

**Comment by mattlewis92** [comment]
Gave some more thought to this:
```
Also the the imported filename needs to respect the output.path and output.publicPath webpack options. output.publicPath is easy as its just publicPath + '/' + filename, output.path is trickier though as it can be a full filepath so you'll need to remove process.cwd() from it. There may be a webpack API / a module to make this easier but I can't find anything from google.
```

and it shouldn't be necessary, as the browser should resolve the file to be the same directory as the main file that imports it.

**Comment by entozoon** [comment]
Ah cool, I was hoping that would be the case but wasn't sure :)
Perhaps we're good to go at this point then?

**Comment by mattlewis92** [comment]
Sorry about the delay on this one, I wanted to flesh out the tests a bit more and found a couple bugs along the way. Released now as 1.1.0, thanks for the PR! 👍

**Comment by entozoon** [comment]
Wicked, no worries! Thanks for accepting and the tips along the way. (Forced me to try chai testing, which seems dead good)

**Pull request merged**